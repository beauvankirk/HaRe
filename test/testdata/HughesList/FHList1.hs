module Main where

interleave :: a -> [a] -> [a]
interleave e lst = (head lst) : e : interleave e (tail lst)

main = putStrLn "hello"
