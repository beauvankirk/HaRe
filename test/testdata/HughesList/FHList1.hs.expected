module FHList1 where
import qualified Data.DList as DList
import Data.DList (DList)

interleave :: a -> [a] -> DList a
interleave e lst = (head lst) `DList.cons` e `DList.cons` interleave e (tail lst)